### PR TITLE
MINOR: [Docs] Use 'console' highlighting (instead of 'shell') for code blocks with $

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
When using `console` highlighting, the `$` automatically is "not selectable" (when selecting for copy-pasting), which doesn't happen with the `shell` highlighting we were using.

Follow-up on https://github.com/apache/arrow/pull/11778#discussion_r759953575

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
cc @AlenkaF

File: docs/source/developers/cpp/building.rst [hunk 1]
```diff
@@ -15,6 +15,8 @@
 .. specific language governing permissions and limitations
 .. under the License.
 
+.. highlight:: console
```
Comment (jorisvandenbossche) [inline docs/source/developers/cpp/building.rst:18]: This sets the default for the full file. Alternative is specifying it for each code-block (like we now had 'shell' for each code block). The latter might actually be more explicit, so either way is fine for me.

**Comment by jorisvandenbossche** [comment]
(I should still check if this can also be done for other files)

File: docs/source/developers/cpp/building.rst (hunk 1 above)
Comment (pitrou) [reply docs/source/developers/cpp/building.rst:18]: Are there other files where we should apply this? (or switch "shell" code blocks to "console"?)

Comment (AlenkaF) [reply docs/source/developers/cpp/building.rst:18]: If there is a need to make a code block in another language I assume it supersedes the `console` set for the full file?

Comment (jorisvandenbossche) [reply docs/source/developers/cpp/building.rst:18]: > If there is a need to make a code block in another language I assume it supersedes the console set for the full file?

Yes, indeed

File: docs/source/conf.py [hunk 2]
```diff
@@ -70,7 +70,7 @@
     'IPython.sphinxext.ipython_directive',
     'IPython.sphinxext.ipython_console_highlighting',
     'breathe',
-    'sphinx_tabs.tabs'
+    # 'sphinx_tabs.tabs'
```
Comment (AlenkaF) [inline docs/source/conf.py:73]: Just curious, did this make trouble that you needed to comment it out?

Comment (jorisvandenbossche) [reply docs/source/conf.py:73]: Whoops, didn't mean to commit that. Yeah, the reason is that I had uninstalled it for updating sphinx (I should reinstall sphinx-tabs with pip, the latest version if not yet available with conda). Will revert this

**Comment by AlenkaF** [comment]
@jorisvandenbossche it could be just my build of the docs again, but `$` gets selected if I try to copy paste. Does it work for u?

**Comment by AlenkaF** [comment]
@jorisvandenbossche probably it's jut my build of the docs again, but trying the `console` directive in another file `$` gets copy-pasted. Did you maybe try this on your build, does it work?

**Pull request closed without merging**